### ISSUE TITLE
`padding-line-between-statements`: add ability to have custom STATEMENT_TYPE

### ISSUE THREAD

**Issue by mjbagheri2545** [eslint-stylistic/eslint-stylistic#1065]
### Clear and concise description of the problem

```ts
export type TPaginationQuery = v.InferOutput<typeof paginationQuerySchema>;
export const paginationQuerySchema = v.object({
  page: v.optional(v.number(), 1),
  limit: v.optional(v.number(), config.query.LIMIT),
});
```

i want to have schema and InferedType toghether, but i can't do this with static STATEMENT_TYPE. there are many cases we can improve coding styles with custom STATEMENT_TYPE, because any developer has her own coding style.

### Suggested solution

```ts
"padding-line-between-statements": [
      "warn",
      { blankLine: "always", prev: ["const", "let", "var"], next: "*" },
      {
        blankLine: "any",
        prev: ["const", "let", "var"],
        next: ["const", "let", "var"],
      },
      { blankLine: "always", prev: ["case", "default"], next: "*" },
      { blankLine: "always", prev: ["type", "interface"], next: "export const *Schema =" },
    ],
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
There was a relevant [issue](https://github.com/eslint-stylistic/eslint-stylistic/issues/956) with some earlier discussion. And feel free to share the implementation you have in mind. If anyone wants to give it a try, PR welcome!
